code/_onclick/hud/new_player.dm

```diff
@@ -348,14 +369,136 @@
+#define SHUTTER_WAIT_DURATION 0.2 SECONDS
+
+/atom/movable/screen/lobby/button/collapse
+	name = "Collapse Lobby Menu"
+	icon = 'icons/hud/lobby/collapse_expand.dmi'
+	icon_state = "collapse"
+	base_icon_state = "collapse"
+	layer = LOBBY_COLLAPSE_BUTTON_LAYER
+	screen_loc = "TOP:-82,CENTER:-54"
+	always_shown = TRUE
+	var/matrix/ce_button_matrix = matrix()
+
+/atom/movable/screen/lobby/button/collapse/Click(location, control, params)
+	. = ..()
+	if(!.)
+		return
+	var/mob/dead/new_player/new_player = hud.mymob
+	base_icon_state = new_player.hud_used.inventory_shown ? "expand" : "collapse"
+	name = "[new_player.hud_used.inventory_shown ? "Collapse" : "Expand"] Lobby Menu"
+	set_button_status(FALSE)
+	//re-enable clicking the button when the shutter animation finishes
+	addtimer(CALLBACK(src, PROC_REF(set_button_status), TRUE), (2 * SHUTTER_MOVEMENT_DURATION + SHUTTER_WAIT_DURATION))
+
+	var/atom/movable/screen/lobby/shutter/menu_shutter = locate(/atom/movable/screen/lobby/shutter) in hud.static_inventory
+	//build a list of bottom buttons for the shutter if it's empty
+	if(!length(menu_shutter.bottom_buttons))
+		for(var/atom/movable/screen/lobby/button/bottom/lobbyscreen in new_player.hud_used.static_inventory)
+			menu_shutter.bottom_buttons += lobbyscreen
+
+	addtimer(CALLBACK(src, PROC_REF(toggle_menu), new_player), SHUTTER_MOVEMENT_DURATION + 1) //ever so slightly after the shutter pulls down
+	//animate the shutter
+	menu_shutter.setup_shutter_animation()
+	//animate bottom buttons' movement
+	if(new_player.hud_used.inventory_shown)
+		menu_shutter.collapse_bottom_buttons(new_player)
+		collapse_button()
+	else
+		menu_shutter.expand_bottom_buttons(new_player)
+		expand_button()
+
+///Moves the button to the top of the screen, leaving only the screen part in view
+/atom/movable/screen/lobby/button/collapse/proc/collapse_button()
+	ce_button_matrix = matrix()
+	//wait for the shutter to come down
+	animate(src, transform = ce_button_matrix, time = SHUTTER_MOVEMENT_DURATION + SHUTTER_WAIT_DURATION)
+	//then pull the button up with the shutter and leave it on the edge of the screen
+	animate(transform = ce_button_matrix.Translate(x = 0, y = 134), time = SHUTTER_MOVEMENT_DURATION, easing = CUBIC_EASING|EASE_IN)
+
+///Extends the button back to its usual spot
+/atom/movable/screen/lobby/button/collapse/proc/expand_button()
+	animate(src, transform = ce_button_matrix.Translate(x = 0, y = -134), time = SHUTTER_MOVEMENT_DURATION, easing = CUBIC_EASING|EASE_OUT)
+	ce_button_matrix = matrix()
+
+///Disables/enables menu buttons on the client's screen
+/atom/movable/screen/lobby/button/collapse/proc/toggle_menu(mob/dead/new_player/new_player)
+	if(new_player.hud_used.inventory_shown)
+		new_player.client.screen -= new_player.hud_used.toggleable_inventory
+	else
+		new_player.client.screen += new_player.hud_used.toggleable_inventory
+	new_player.hud_used.inventory_shown = !new_player.hud_used.inventory_shown
```

Comment:
typecast-capture the datum for `hud_used` so we don't have to keep traversing all the way through new_player five times.

Reply:
https://github.com/tgstation/tgstation/pull/76443/commits/0fb8e4ac86e78604a77158dd1b4b2aaf099c0b82